src/app/shared/cms/cms.service.ts

```diff
@@ -0,0 +1,54 @@
+import {Injectable} from 'angular2/core';
+import {Observable} from 'rxjs/Observable';
+import {ReplaySubject} from 'rxjs';
+import {Http} from 'angular2/http';
+import {HalDoc} from './haldoc';
+import {Env} from '../../../util/env';
+
+@Injectable()
+export class CmsService {
+
+  public authToken: ReplaySubject<string>;
+  private rootDoc: ReplaySubject<any>;
+
+  constructor(private http: Http) {
+    this.authToken = new ReplaySubject(1);
+    this.rootDoc = new ReplaySubject(1);
+    this.http.get(`${Env.CMS_HOST}/api/v1`).subscribe((res) => {
+      this.rootDoc.next(res.json());
+    });
+  }
+
+  set token(token: string) {
+    this.authToken.next(token);
+  }
+
+  get root(): Observable<HalDoc> {
```

Comment:
could be premature. Just thinking other hal services will end up with similar methods.